### DISCUSSION TITLE
Visual Studio for Mac: Retirement & Migration

### DISCUSSION THREAD

**Discussion by projectgoav** [actions/virtual-environments#8212] [General questions]
### Description

Microsoft have announced that Visual Studio for Mac will be discontinued in the near future: https://devblogs.microsoft.com/visualstudio/visual-studio-for-mac-retirement-announcement/

- Will Pipelines / Github Actions still provide images with the .NET iOS & Android workloads installed if they are no longer provided by VS4Mac?
- Will .NET iOS & .NET Android MAUI and native apps (using the Java/UIKit APIs) still be supported by Pipelines / Github Actions and the various provided build steps?
- What, if any, changes will we need to make in order to continue building like we currently do?
- Is there a sunset plan for when VS4Mac will cease to be provided by Pipelines / Github Actions? 

### Platforms affected

- [X] Azure DevOps
- [X] GitHub Actions - Standard Runners
- [X] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [X] macOS 11
- [X] macOS 12
- [X] macOS 13
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

All currently provided images with VS4Mac pre-installed

### Is it regression?

Yes - a required tool is being discontinued without much information on the build side of things

### Expected behavior

N/A

### Actual behavior

N/A

### Repro steps

N/A

**Comment by vpolikarpov-akvelon** [comment]
Hey @projectgoav, thank you for highlighting, we aware of these changes, but we're not ready to make any announcement yet

**Comment by projectgoav** [comment]
Any idea on when an announcement might be made? 

There was some talk of VS4Mac continued to being supported via Piplines & Github actions in the comments of the original announcement but unless you are taking over the development of VS4Mac I don't think that's possible.

**Comment by projectgoav** [comment]
Disappointing you never got around to making any announcement about VS4Mac and it's support in Pipelines.

**Reply by Cheesebaron** [reply]
Migrate to .NET 9. Install android and ios workloads, not much else to do to migrate. What specific information are you seeking?

**Comment by Junjun13910** [comment]
Can someone i can ask in telegram because i was trying ti past my hash but on the side said I don't need to submit pls answer thanks

**Comment by Junjun13910** [comment]
Juspay

Adrian
